src/charm_state.py

```diff
@@ -266,6 +268,45 @@ def userids_to_list(cls, value: str) -> typing.List[str]:
                 raise ValidationError(f"Invalid user ID format: {user_id}", cls)
         return value_list
 
+    @validator("experimental_alive_check")
+    @classmethod
+    def to_pebble_check(cls, value: str) -> typing.Dict[str, typing.Union[str, int]]:
+        """Convert the experimental_alive_check field to pebble check.
+
+        Args:
+            value: the input value.
+
+        Returns:
+            The pebble check.
+
+        Raises:
+            ValidationError: if experimental_alive_check is invalid.
+        """
+        # expected
+        # period,threshold,timeout
+        config_values = value.split(",")
+        if len(config_values) != 3:
+            raise ValidationError(
+                f"Invalid experimental_alive_check, less or more than 3 values: {value}", cls
+            )
+        try:
+            period = config_values[0].strip().lower()
+            if period[-1] not in ("s", "m", "h"):
+                raise ValidationError(
+                    f"Invalid experimental_alive_check, period should end in s/m/h: {value}", cls
+                )
+            threshold = int(config_values[1].strip())
+            timeout = config_values[2].strip().lower()
+            if timeout[-1] not in ("s", "m", "h"):
```

Comment:
Is there a default for the units, e.g. `15` meaning `15m` or `15s`?

cc @benhoyt as I can't make it out from documentation.

Reply:
No, there's no default -- they require a unit suffix for clarity. The format of these intervals is Go's `time.Duration` format, documented briefly here: https://pkg.go.dev/time#ParseDuration

Yeah, we should clear that up in our docs. Docs are in flux right now with Tiexin moving them to a docs site, but feel free to open an issue.